components/mailbox/src/Mailbox.svelte

```diff
@@ -392,6 +393,27 @@
     returnToMailbox();
   }
 
+  async function downloadSelectedFile(event: CustomEvent) {
+    const file = event.detail.file;
+    const downloadedFileData = await downloadFile({
+      file_id: file.id,
+      component_id: id,
+      access_token,
+    });
+    const buffer = Uint8Array.from(atob(downloadedFileData), (c) =>
+      c.charCodeAt(0),
+    );
+    var blob = new Blob([buffer], { type: file.content_type });
+    var blobFile = window.URL.createObjectURL(blob);
+
+    var a = document.createElement("a");
```

Comment:
```suggestion
    const blob = new Blob([buffer], { type: file.content_type });
    const blobFile = window.URL.createObjectURL(blob);

    const a = document.createElement("a");
```

Reply:
This code is also repeated in the `Email.svelte` component, could we move it to commons and import it?

Reply:
That's a good idea, will do